packages/xod-client/src/core/containers/App.jsx

```diff
@@ -74,12 +79,48 @@ export default class App extends React.Component {
     )();
   }
 
-  transformProjectForTranspiler(debug = false) {
+  onRunSimulation() {
+    if (this.props.isSimulationRunning) {
+      this.props.actions.addError(SIMULATION_ALREADY_RUNNING);
+      return;
+    }
+    this.props.actions.runSimulationRequested();
+    const eitherTProject = this.transformProjectForTranspiler(false, true);
+
+    R.compose(
+      foldEither(
+        R.compose(
+          this.props.actions.addError,
+          R.when(R.is(Error), formatErrorMessage)
+        ),
+        ({ code, nodeIdsMap }) =>
+          this.props.actions.runSimulation(
+            explodeMaybe(
+              'currentPatchPath already folded in `transformProjectForTranspiler`',
+              this.props.currentPatchPath
+            ),
+            nodeIdsMap,
+            code
+          )
+      ),
+      R.map(
+        R.applySpec({
+          code: transpile,
+          nodeIdsMap: getNodeIdsMap,
+        })
+      )
+    )(eitherTProject);
+  }
+
+  transformProjectForTranspiler(debug = false, simulation = false) {
```

Comment:
Please, don’t use positional args defaults. They complicate currying and code reasoning.